### PR TITLE
Force encoding in reading of setup.cfg to utf-8

### LINKED ISSUE #4328: PyLint Fails when there are non-ascii characters in setup.cfg
Hi there,

I get the following error, when running pylint from spyder (on a single file):
```bash
Traceback (most recent call last):
  File "/usr/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/usr/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/usr/local/lib/python3.6/dist-packages/pylint/__main__.py", line 18, in <module>
    pylint.run_pylint()
  File "/usr/local/lib/python3.6/dist-packages/pylint/__init__.py", line 19, in run_pylint
    from pylint.lint import Run as PylintRun
  File "/usr/local/lib/python3.6/dist-packages/pylint/lint/__init__.py", line 76, in <module>
    from pylint.lint.pylinter import PyLinter
  File "/usr/local/lib/python3.6/dist-packages/pylint/lint/pylinter.py", line 19, in <module>
    from pylint import checkers, config, exceptions, interfaces, reporters
  File "/usr/local/lib/python3.6/dist-packages/pylint/checkers/__init__.py", line 45, in <module>
    from pylint.checkers.base_checker import BaseChecker, BaseTokenChecker
  File "/usr/local/lib/python3.6/dist-packages/pylint/checkers/base_checker.py", line 20, in <module>
    from pylint.config import OptionsProviderMixIn
  File "/usr/local/lib/python3.6/dist-packages/pylint/config/__init__.py", line 103, in <module>
    PYLINTRC = find_pylintrc()
  File "/usr/local/lib/python3.6/dist-packages/pylint/config/__init__.py", line 96, in find_pylintrc
    for config_file in find_default_config_files():
  File "/usr/local/lib/python3.6/dist-packages/pylint/config/find_default_config_files.py", line 35, in find_default_config_files
    if config_name.endswith(".cfg") and not _cfg_has_config(config_name):
  File "/usr/local/lib/python3.6/dist-packages/pylint/config/find_default_config_files.py", line 23, in _cfg_has_config
    parser.read(path)
  File "/usr/lib/python3.6/configparser.py", line 697, in read
    self._read(fp, filename)
  File "/usr/lib/python3.6/configparser.py", line 1015, in _read
    for lineno, line in enumerate(fp, start=1):
  File "/usr/lib/python3.6/encodings/ascii.py", line 26, in decode
    return codecs.ascii_decode(input, self.errors)[0]
UnicodeDecodeError: 'ascii' codec can't decode byte 0xc3 in position 227: ordinal not in range(128)
```
Changing this line:
https://github.com/PyCQA/pylint/blob/5e1928b325bc798f5be1ab94031bf6816d058d9f/pylint/config/find_default_config_files.py#L29
to
```python
parser.read(path, encoding='utf-8')
```
makes it at least working. I don't know the rational behind this or how to force pylint to read setup.cfg encoded as `utf-8`. Putting `# -*- coding: utf-8 -*-` on top of the file didn't help. Or should I prevent umlauts in `setup.cfg`?!

Am I missing something?

Cheers,
Sebastian

### PR DESCRIPTION
**Pull request by MuellerSeb**
## Steps

- [x] Add yourself to CONTRIBUTORS if you are a new contributor.
- [x] Add a ChangeLog entry describing what your PR does.
- [ ] If it's a new feature or an important bug fix, add a What's New entry in
      `doc/whatsnew/<current release.rst>`.
- [x] Write a good description on what the PR does.

## Description

Pylint crashed when trying to read setup.cfg with non-ascii characters.

## Type of Changes

Force encoding to be `"utf-8"` in `_cfg_has_config`.

|     | Type                   |
| --- | ---------------------- |
| ✓   | :bug: Bug fix          |
|     | :sparkles: New feature |
|     | :hammer: Refactoring   |
|     | :scroll: Docs          |

## Related Issue

Closes #4328

### CODE REVIEW

**Comment by MuellerSeb** [comment]
I can't tell if the failing pre-commit check is important or not. Otherwise this PR is ready for review from my side.

File: ChangeLog [hunk 1]
```diff
@@ -26,6 +26,9 @@ Release date: Undefined
 
   Closes #4301
 
+* Fix crash when checking ``setup.cfg`` for pylint config when there are non-ascii characters in there
+
+  Closes #4328
```
Comment (cdce8p) [inline ChangeLog:31]: You probably need to add an empty line here (after `Closes ...`). If you like, you can do `pre-commit install` which will install git pre commit hooks that automatically check for such issues.

**Review by Pierre-Sassoulas — APPROVED** [review]
Thanks for troubleshooting this and opening a pull request !

File: ChangeLog (hunk 1 above)
Comment (MuellerSeb) [reply ChangeLog:31]: Done.

File: ChangeLog [hunk 2]
```diff
@@ -26,6 +26,10 @@ Release date: Undefined
...
```
Comment (cdce8p) [inline ChangeLog:31]: Sorry for not noticing it earlier, I just saw that this should belong to `2.7.5` not `2.8.0`.
Would you like to change it if not I can push a commit as well.

Comment (MuellerSeb) [reply ChangeLog]: Done.

**Comment by Pierre-Sassoulas** [comment]
Thank you for fixing this :)

**Pull request merged**